File: ITing-frontend/src/services/adminPermissionService.js
import axiosInstance from "../utils/axiosInstance";

const adminPermissionService = {
  getOverrides: async (userId) => {
    const response = await axiosInstance.get(`/admin/permissions/users/${userId}/overrides`);
    return response;
  },

  replaceOverrides: async (userId, overrides) => {
    const response = await axiosInstance.put(`/admin/permissions/users/${userId}/overrides`, overrides);
    return response;
  },

  clearOverrides: async (userId) => {
    const response = await axiosInstance.delete(`/admin/permissions/users/${userId}/overrides`);
    return response;
  },
};

export default adminPermissionService;

File: ITing-backend/src/main/java/com/iting/jobportal/admin/service/AdminPermissionService.java
package com.iting.jobportal.admin.service;

import java.util.List;
import java.util.Map;

public interface AdminPermissionService {
    Map<String, Boolean> getOverrides(Long accountId);
    void replaceOverrides(Long adminId, Long accountId, Map<String, Boolean> overrides);
    void deleteOverride(Long adminId, Long accountId, String permissionKey);
    void clearOverrides(Long adminId, Long accountId);
    /** Apply the same set of overrides to multiple users at once. */
    void bulkReplaceOverrides(Long adminId, List<Long> accountIds, Map<String, Boolean> overrides);
}
